### PR TITLE
Updating event api docs to include event.detail.

### PR DESCRIPTION
**Pull request by internettrans**

### CODE REVIEW

File: website/package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "rename-version": "docusaurus-rename-version"
   },
   "devDependencies": {
-    "docusaurus": "^1.2.1",
+    "docusaurus": "^1.7.2",
```
Comment (internettrans) [inline website/package.json:12]: blind trust in docusaurus' semantic versioning

File: .circleci/config.yml [hunk 2]
```diff
@@ -4,9 +4,6 @@ jobs:
     docker:
       # specify the version you desire here
       - image: circleci/node:8.11.1
-    branches:
```
Comment (internettrans) [inline .circleci/config.yml:7]: this didn't work. It turned off all deployments, including master branch.

File: .circleci/config.yml [hunk 3]
```diff
@@ -20,4 +17,7 @@ workflows:
   version: 2
   build_and_deploy:
     jobs:
-      - deploy-website
+      - deploy-website:
```
Comment (internettrans) [inline .circleci/config.yml:20]: See https://discuss.circleci.com/t/triggering-build-on-changes-to-only-one-branch-when-using-a-workflow/20430/3

File: docs/api.md [hunk 4]
```diff
@@ -122,42 +122,52 @@ The mounted parcel will not be automatically unmounted and unmounting will need
...
 This event will get fired after each hashchange, popstate, or triggerAppChange, even if no changes
-to registered applications were necessary. Sample usage of this event might look like this:
+to registered applications were necessary. The `event.detail` for the event is the native DOM event that triggered the reroute
+(such as a [PopStateEvent](https://developer.mozilla.org/en-US/docs/Web/API/PopStateEvent) or [HashChangeEvent](https://developer.mozilla.org/en-US/docs/Web/API/HashChangeEvent)).
+Sample usage of this event might look like this:
 ```js
-window.addEventListener('single-spa:before-routing-event', () => {
-	console.log('before routing event occurred!');
+window.addEventListener('single-spa:before-routing-event', evt => {
+  const originalEvent = evt.detail
```
Comment (internettrans) [inline docs/api.md:130]: See https://github.com/CanopyTax/single-spa/releases/tag/v4.0.2

File: website/package.json (hunk 1 above)
Comment (frehner) [reply website/package.json:12]: https://github.com/facebook/Docusaurus/blob/master/CHANGELOG.md#130---2018-06-20

I haven't had time to look into if/how this affected us, but it was what was preventing me from blindly trusting them :/

**Review by frehner — COMMENTED** [review]
one comment about docusaurus semver

**Comment by internettrans** [comment]
I think that those breaking changes are fine -- looked at them in local testing and didn't see anything

**Review by frehner — APPROVED** [review]

**Pull request merged**